### DISCUSSION TITLE
filter a collection to only include items which *don't* have items associated in a m2m field

### DISCUSSION THREAD

**Discussion by wc-matteo** [directus/directus#9858] [Archive]
I need to find the items in a collection which don't have any related items in a m2m field.

But I can't seem to find a way.

I can find the items which _do_ have related items, but not the ones without related items.

Might be related to [this](https://github.com/directus/directus/discussions/7931#discussioncomment-1376019) comment.

Any ideas/workarounds?

**Comment by wc-matteo** [comment]
@rijkvanzanten can you confirm this is not possible atm?

**Reply by rijkvanzanten** [reply]
That's correct. There's no "count" modifier/function to filter against the amount of related items in a o2m/m2m field at the moment

**Comment by azrikahar** [comment]
This is now made possible by #12488! So you can do something like `?filter[count(your_related_field)][_eq]=0` for ones _without_ related items.

**Reply by ridwankustanto** [reply]
What if the array that I want to count is in each array object nested from the parent field? I tried the same approach but had no luck.

example:

    {
        "category": "ABC",
        "articles": [
            {
                "name": "Article 1",
                "likes": [
                    "user_id_1",
                    "user_id_2",
                    ...
                ]
            }
        ]
    }
    
    i tried `&fields=count(articles.likes)`. but it doesn't work. Any idea?

**Reply by rijkvanzanten** [reply]
`count` like that is done through the `aggregate` query param: https://docs.directus.io/reference/query.html#aggregation-grouping 👍🏻
